### DISCUSSION TITLE
Feature Request: Support INSERT OR REPLACE Statement

### DISCUSSION THREAD

**Discussion by murraybauer** [duckdb/duckdb#5957] [Ideas]
Why do you want this feature?

It is a common workflow to insert new data into the database where by some values allready exist and need to be updated instead.
Avoids needing to delete the overlapping data data first, then do a insert.

**Syntax**
```
INSERT OR REPLACE INTO <table_name> [(<column_name_list>)] VALUES (<column_value_list>)[,(<column_value_list>)]...
```

You can use the INSERT OR REPLACE statement to write new rows or replace existing rows in the table. 
The syntax and behavior of the INSERT OR REPLACE statement is similar to the INSERT statement. 
Unlike the INSERT statement, the INSERT OR REPLACE statement does not generate an error if a row already exists.

https://docs.tibco.com/pub/as/4.2.0/doc/html/GUID-8DAD3152-EBD3-46DB-B724-397D0AF0C94B.html

**Comment by Mytherin** [comment]
There is a PR open that implements this feature, see #5866
